apps/web/src/views/Farms/hooks/getNonBscVaultFee.ts

```diff
@@ -79,8 +81,14 @@ export const getNonBscVaultContractFee = async ({
         userAddress as Address,
         messageType,
       ])
+      const publicClient = getPublicClient({ chainId })
+      const { maxFeePerGas } = await estimateFeesPerGas(publicClient)
+      const evmGasPrice = maxFeePerGas?.toString()
+      if (!evmGasPrice) {
+        throw new Error('Missing evm gasPrice')
+      }
```

Comment:
can we fix this in the `useGasPrice` hook?

Reply:
ideally the hook should accept `chainId` as input and derive the correct gas price on target chain

Reply:
My thought is to have a unified `useGasPrice` hook for all chains. Can we specify a list of chains that follow eip 1559? Like if the requested chain falls in this list then we'll calculate using maxFeePerGas, otherwise we'll fallback to the default way

Reply:
ok I think we can use this approach for now. One more concern is that, it will break if we have v2 farms on other non bsc chain. Since we might not be able to get `maxFeePerGas`. 

Can we use `maxFeePerGas` as a fallback? We can use the default gasPrice if there's a valid input
```typescript
const evmGasPrice =  gasPrice ?? maxFeePerGas?.toString()
```

Secondly, just realized that the `useFeeData` hook from `@pancakeswap/wagmi` is broken. Could you help fix as well?